stan/math/prim/mat/meta/container_view.hpp

```diff
@@ -0,0 +1,208 @@
+#ifndef STAN_MATH_PRIM_MAT_META_CONTAINER_VIEW_HPP
+#define STAN_MATH_PRIM_MAT_META_CONTAINER_VIEW_HPP
+
+#include <stan/math/prim/scal/meta/container_view.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <vector>
+
+namespace stan {
+
+  namespace math {
+
+    /**
+     * Template specialization for column vector view of
+     * array y with scalar type T2 with size inferred from
+     * input column vector x     
+     *
+     * operator[](int i) returns reference to view 
+     * broadcasts as if x is vector<Matrix>
+     *
+     * Intended for use in OperandsAndPartials
+     *
+     * @tparam T1 scalar type of input matrix
+     * @tparam T2 scalar type returned by view.
+     * @param x input matrix
+     * @param y underlying array 
+     */
+
+    template <typename T1, typename T2>
+    class container_view<Eigen::Matrix<T1, -1, 1>, Eigen::Matrix<T2, -1, 1> > {
```

Comment:
Most of our code uses `Eigen::Dynamic` instead of `-1` and I think we should be consistent.  

I guess at this point, I'd be OK globally replacing `Eigen::Dynamic` with `-1`